src/matcher.ts

```diff
@@ -0,0 +1,88 @@
+import { computeSimilarity } from "./similarity";
+import { Headers, TapeRecord } from "./tape";
+
+/**
+ * Returns the first of a list of records that hasn't been replayed before.
+ *
+ * This is useful to ensure that when multiple requests to the same endpoint
+ * have been recorded, each of them is replayed in the same order.
+ *
+ * If there are more requests to be replayed than we have recorded, this will
+ * always pick the last one.
+ *
+ * @param records A list of records (e.g. returned by findRecordMatches).
+ * @param tapeReplayCount A set of tape records that have been replayed before.
+ */
+export function findFirstLeastUsedRecord(
+  records: TapeRecord[],
+  replayedTapes: Set<TapeRecord>
+): TapeRecord | null {
+  // Look for a record that hasn't been replayed yet.
+  for (const record of records) {
+    if (!replayedTapes.has(record)) {
+      replayedTapes.add(record);
```

Comment:
It's a bit confusing if the purpose of this function is to _"find"_ yet it updates the set given to it.